src/runtime/HalideBuffer.h

```diff
@@ -150,16 +150,17 @@ class Buffer {
         return halide_type_of<typename std::remove_cv<not_void_T>::type>();
     }
 
-    /** Is this Buffer responsible for managing its own memory */
-    bool manages_memory() const {
+    /** Does this Buffer own the host memory it refers to? */
+    bool owns_host_memory() const {
         return alloc != nullptr;
     }
 
 private:
     /** Increment the reference count of any owned allocation */
     void incref() const {
-        if (!manages_memory()) return;
-        alloc->ref_count++;
+        if (owns_host_memory()) {
+            alloc->ref_count++;
+        }
         if (buf.device) {
```

Comment:
It won't be, but decref_dev will now be called unconditionally by decref, which will notice that there's a non-zero device field and free it.